### ISSUE TITLE
Another color of power lasers, when energy comes from batteries

### ISSUE THREAD

**Issue by Qvertyes** [Anuken/Mindustry-Suggestions#2012]
This will help in time to notice the explosion of the reactor or its shutdown. This feature in general will help to control the amount of consumed and produced energy. 

1. - [x] I have done a quick search in the list of suggestions to make sure this has not been suggested yet.
2. - [x] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
3. - [x] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
4. - [x] I have read `README.md` to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by joshuaptfan** [comment]
What you mean is, "change color of power lasers when power is negative".

**Comment by Kyllingene** [comment]
1. please do what the boxes say
2. after doing 1., fill them in
3. i see no reason to not do this

**Comment by Qvertyes** [comment]
> What you mean is, "change color of power lasers when power is negative".
Power lasers between power nodes don't change color, when batteries have charge and energy goes into negative

**Comment by Qvertyes** [comment]
>     1. please do what the boxes say
>     2. after doing 1., fill them in
>     3. i see no reason to not do this

Sorry, i don't understand what you mean, my English so bad

**Comment by joshuaptfan** [comment]
It's ok, we understand what you mean. Maybe I'll work on a PR for this.

**Comment by Kyllingene** [comment]
Ah, then that's OK.

**Comment by sup3r87** [comment]
actually a pretty good idea! i like this

**Comment by WinterCrystal** [comment]
I like this idea and it feels like something Anuke should've done a long time ago. It helps people distinguish whether certain areas are underpowered or have no power at all.

**Comment by itcannotbe** [comment]
Actually ........
![Screenshot_20210222_082510](https://user-images.githubusercontent.com/45280919/108643916-9c17b980-74e7-11eb-9f2c-7395799720f0.jpg)

**Comment by Kyllingene** [comment]
Yes, however, there are two problems with the current system:
1. Battery power and no power are the same
2. The colors are quite similar and hard to distinguish for some people

**Comment by itcannotbe** [comment]
I've personally yet to encounter people complaing that they can't distinguish white from yellow.

Also battery power is a different problem altogether.

**Comment by Kyllingene** [comment]
> white from yellow.

1. yellowish-white*
2. i have met such people

**Comment by itcannotbe** [comment]
Actually the center of the power laser beam is white.

**Comment by Kyllingene** [comment]
And it turns to yellow at the edges.

**Comment by itcannotbe** [comment]
So enlarge the white portion?

**Comment by Kyllingene** [comment]
How? There are only a couple pixels left, so enlarging the white would make the laser look like a solid white line (because it would be). Possibly a power laser size slider, along with a different color for battery power?

**Comment by itcannotbe** [comment]
Actually the white portion is above the base yellow laser.

**Comment by Kyllingene** [comment]
I know, what I meant is that there are only a few pixels you could increase the white sections size by before it covered the yellow bit.

**Comment by Yell245** [comment]
Maybe indicate it with red-yellow pulse meaning that "hey energy is negative and buildings will be unpowered soon"

**Comment by Qvertyes** [comment]
> Maybe indicate it with red-yellow pulse meaning that "hey energy is negative and buildings will be unpowered soon"

good idea, i agree

**Comment by Kyllingene** [comment]
^^^

**Comment by joshuaptfan** [comment]
Anyone who supports this suggestion should provide feedback/support to [my pull request implementing this](https://github.com/Anuken/Mindustry/pull/4820).

> Maybe indicate it with red-yellow pulse meaning that "hey energy is negative and buildings will be unpowered soon"

I did not use pulsing color because in the situation where you have fluctuating power production and are rapidly alternating between positive and negative net power, any pulsing effect would not provide you a clear picture of what is happening. So I opted for a solid reddish color instead. Also, pulsing/blinking would be annoying.
